src/test/java/uk/ac/ebi/eva/pipeline/parameters/validation/job/AnnotationJobParametersValidatorTest.java

```diff
@@ -0,0 +1,134 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.springframework.batch.core.JobParameter;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.JobParametersInvalidException;
+
+import uk.ac.ebi.eva.pipeline.parameters.JobParametersNames;
+import uk.ac.ebi.eva.test.rules.PipelineTemporaryFolderRule;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Tests that the arguments necessary to run a {@link uk.ac.ebi.eva.pipeline.jobs.GenotypedVcfJob} are
+ * correctly validated
+ */
+public class AnnotationJobParametersValidatorTest {
+
+    private AnnotationJobParametersValidator validator;
+
+    @Rule
+    public PipelineTemporaryFolderRule temporaryFolder = new PipelineTemporaryFolderRule();
+
+    private Map<String, JobParameter> requiredParameters;
+
+    private Map<String, JobParameter> optionalParameters;
+
+    @Before
+    public void setUp() throws Exception {
+        validator = new AnnotationJobParametersValidator();
+        final String dir = temporaryFolder.getRoot().getCanonicalPath();
+
+        requiredParameters = new TreeMap<>();
+
+        requiredParameters.put(JobParametersNames.DB_NAME, new JobParameter("eva_testing"));
+        requiredParameters.put(JobParametersNames.DB_COLLECTIONS_VARIANTS_NAME, new JobParameter("variants"));
+        requiredParameters.put(JobParametersNames.OUTPUT_DIR_ANNOTATION, new JobParameter(dir));
+        requiredParameters.put(JobParametersNames.APP_VEP_CACHE_SPECIES, new JobParameter("Human"));
+        requiredParameters.put(JobParametersNames.APP_VEP_CACHE_VERSION, new JobParameter("100_A"));
+        requiredParameters.put(JobParametersNames.APP_VEP_VERSION, new JobParameter("80"));
+        requiredParameters.put(JobParametersNames.APP_VEP_NUMFORKS, new JobParameter("6"));
+        requiredParameters.put(JobParametersNames.APP_VEP_TIMEOUT, new JobParameter("600"));
+        requiredParameters.put(JobParametersNames.ANNOTATION_OVERWRITE, new JobParameter("false"));
+        requiredParameters.put(JobParametersNames.DB_COLLECTIONS_ANNOTATION_METADATA_NAME,
+                new JobParameter("annotationMetadata"));
+        requiredParameters.put(JobParametersNames.APP_VEP_CACHE_PATH,
+                new JobParameter(temporaryFolder.getRoot().getCanonicalPath()));
+        requiredParameters.put(JobParametersNames.APP_VEP_PATH,
+                new JobParameter(temporaryFolder.newFile().getCanonicalPath()));
+        requiredParameters.put(JobParametersNames.INPUT_FASTA,
+                new JobParameter(temporaryFolder.newFile().getCanonicalPath()));
+
+        // optionals
+        optionalParameters = new TreeMap<>();
+        optionalParameters.put(JobParametersNames.CONFIG_CHUNK_SIZE, new JobParameter("100"));
+        optionalParameters.put(JobParametersNames.CONFIG_RESTARTABILITY_ALLOW, new JobParameter("true"));
+        optionalParameters.put(JobParametersNames.STATISTICS_OVERWRITE, new JobParameter("true"));
```

Comment:
This is not going to fail, but statistics related parameters (overwrite and skip) shouldn't be really provided.